extensions/amp-image-slider/1.0/component.js

```diff
@@ -0,0 +1,607 @@
+  }, []);
+
+  /**
+   * One step left
+   * @param {boolean=} opt_toEnd
+   * @private
+   */
+  const stepLeft = useCallback(
+    (opt_toEnd) => {
+      // To the very end of left
+      if (opt_toEnd === true) {
+        updatePositions(0);
+      } else {
+        const newPercentage = limitPercentage(
+          getCurrentSliderPercentage() - stepSize
+        );
+        updatePositions(newPercentage);
+      }
+    },
+    [getCurrentSliderPercentage, updatePositions, limitPercentage, stepSize]
+  );
+
+  /**
+   * Step to the center
+   * @private
+   */
+  const stepExactCenter = useCallback(() => {
+    updatePositions(0.5);
+  }, [updatePositions]);
+
+  /**
+   * One step right
+   * @param {boolean=} opt_toEnd
+   * @private
+   */
+  const stepRight = useCallback(
+    (opt_toEnd) => {
+      // To the very end of right
+      if (opt_toEnd === true) {
+        updatePositions(1);
+      } else {
+        const newPercentage = limitPercentage(
+          getCurrentSliderPercentage() + stepSize
+        );
+        updatePositions(newPercentage);
+      }
+    },
+    [updatePositions, limitPercentage, getCurrentSliderPercentage, stepSize]
+  );
+
+  /**
+   * Handler on key down
+   * @param {Event} e
+   * @private
+   */
+  const onKeyDown = useCallback(
+    (e) => {
+      // Check if current element has focus
+      if (
+        doc.current.activeElement !== containerRef?.current &&
+        doc.current.activeElement !== containerRef?.current?./*OK*/ offsetParent
+      ) {
+        return;
+      }
+
+      animateHideHint();
+
+      switch (e.key.toLowerCase()) {
+        case 'left': // Edge non-standard, pre EdgeHTML 17
+        case 'arrowleft':
+          e.preventDefault();
+          e.stopPropagation();
+          stepLeft();
+          break;
+        case 'right': // Edge non-standard, pre EdgeHTML 17
+        case 'arrowright':
+          e.preventDefault();
+          e.stopPropagation();
+          stepRight();
+          break;
+        case 'pageup':
+          // prevent scrolling the page
+          e.preventDefault();
+          e.stopPropagation();
+          stepLeft(true);
+          break;
+        case 'pagedown':
+          e.preventDefault();
+          e.stopPropagation();
+          stepRight(true);
+          break;
+        case 'home':
+          e.preventDefault();
+          e.stopPropagation();
+          stepExactCenter();
+          break;
+      }
+    },
+    [animateHideHint, stepLeft, stepRight, stepExactCenter]
+  );
+  /**
+   * Handler on mouse move
+   * @param {Event} e
+   * @private
+   */
+  const onMouseMove = useCallback(
+    (e) => {
+      e.preventDefault();
+      pointerMoveX(e.pageX);
+    },
+    [pointerMoveX]
+  );
+
+  /**
+   * Handler on mouse button up
+   * @param {Event} e
+   * @private
+   */
+  const onMouseUp = useCallback(
+    (e) => {
+      e.preventDefault();
+      unlisten(unlistenMouseMove?.current);
+      unlisten(unlistenMouseUp?.current);
```

Comment:
I suspect this isn't working as intended.

Reply:
This should be resolved with https://github.com/ampproject/amphtml/pull/35783/commits/dec64867137046caa24c417b5f8c1b6af6c07262